### PR TITLE
*: add a new variable to control whether to invalidate plan cache when related table analyzed

### LINKED ISSUE #43295: Commit 86f215 cause a 20% performance regression of pacer workload
## Bug Report

Please answer these questions before submitting your issue. Thanks!

### 1. Minimal reproduce step (Required)
Deploy a tidb cluster with commit 86f215d8f593996c3593ac866dd2fea2846eda41 and run pacer workload, the plan of the following SQL changed from `index lookup` to `table scan` the QPS drops 20% comparing to commit a20e7fd3a462c4ae122f7b65d60e89e4c65a7b39

```
SELECT
local_id,
state,
attempts_allowed,
attempts_remaining,
body,
custom_status,
updated_at,
created_at,
run_after,
server_status
FROM
pacerq_pinlater_pacer_benchmark_empty_job_tidb_37
WHERE
state = ?
AND run_after <= ?
LIMIT
? [arguments: (1, "2023-04-13 10:24:44.000000", 10)];
```

<!-- a step by step guide for reproducing the bug. -->

### 2. What did you expect to see? (Required)
```
	id                          	task     	estRows	operator info                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                     	actRows	execution info                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                    	memory 	disk
	Projection_7                	root     	10     	pacer.pacerq_pinlater_pacer_benchmark_empty_job_tidb_5.local_id, pacer.pacerq_pinlater_pacer_benchmark_empty_job_tidb_5.state, pacer.pacerq_pinlater_pacer_benchmark_empty_job_tidb_5.attempts_allowed, pacer.pacerq_pinlater_pacer_benchmark_empty_job_tidb_5.attempts_remaining, pacer.pacerq_pinlater_pacer_benchmark_empty_job_tidb_5.body, pacer.pacerq_pinlater_pacer_benchmark_empty_job_tidb_5.custom_status, pacer.pacerq_pinlater_pacer_benchmark_empty_job_tidb_5.updated_at, pacer.pacerq_pinlater_pacer_benchmark_empty_job_tidb_5.created_at, pacer.pacerq_pinlater_pacer_benchmark_empty_job_tidb_5.run_after, pacer.pacerq_pinlater_pacer_benchmark_empty_job_tidb_5.server_status	10     	time:717.1ms, loops:2, Concurrency:OFF                                                                                                                                                                                                                                                                                                                                                                                                                                                                                            	4.71 KB	N/A
	└─IndexLookUp_17            	root     	10     	limit embedded(offset:0, count:10)                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                	10     	time:717.1ms, loops:2, index_task: {total_time: 716.4ms, fetch_handle: 716.4ms, build: 1µs, wait: 4.64µs}, table_task: {total_time: 667µs, num: 1, concurrency: 5}, next: {wait_index: 716.4ms, wait_table_lookup_build: 46.7µs, wait_table_lookup_resp: 622.4µs}                                                                                                                                                                                                                                                            	12.9 KB	N/A
	  ├─Limit_16(Build)         	cop[tikv]	10     	offset:0, count:10                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                	10     	time:716.4ms, loops:1, cop_task: {num: 1, max: 716.3ms, proc_keys: 10, tot_proc: 91.7ms, tot_wait: 53.2µs, rpc_num: 1, rpc_time: 716.2ms, copr_cache_hit_ratio: 0.00, build_task_duration: 17.5µs, max_distsql_concurrency: 1}, tikv_task:{time:92ms, loops:1}, scan_detail: {total_process_keys: 10, total_process_keys_size: 550, total_keys: 153531, get_snapshot_time: 33.1µs, rocksdb: {delete_skipped_count: 51930, key_skipped_count: 205460, block: {cache_hit_count: 283}}}                                           	N/A    	N/A
	  │ └─IndexRangeScan_14     	cop[tikv]	10     	table:pacerq_pinlater_pacer_benchmark_empty_job_tidb_5, index:run_after(state, run_after), range:[1 -inf,1 2023-04-11 03:44:07], keep order:false, stats:pseudo                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                   	10     	tikv_task:{time:92ms, loops:1}                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                    	N/A    	N/A
	  └─TableRowIDScan_15(Probe)	cop[tikv]	10     	table:pacerq_pinlater_pacer_benchmark_empty_job_tidb_5, keep order:false, stats:pseudo                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                            	10     	time:610.7µs, loops:2, cop_task: {num: 1, max: 567µs, proc_keys: 10, tot_proc: 198.6µs, tot_wait: 24.5µs, rpc_num: 1, rpc_time: 548µs, copr_cache_hit_ratio: 0.00, build_task_duration: 8.25µs, max_distsql_concurrency: 1, max_extra_concurrency: 1}, tikv_task:{time:0s, loops:1}, scan_detail: {total_process_keys: 10, total_process_keys_size: 1410, total_keys: 11, get_snapshot_time: 5.35µs, rocksdb: {key_skipped_count: 10, block: {cache_hit_count: 23, read_count: 2, read_byte: 31.0 KB, read_time: 34.8µs}}}	N/A    	N/A
```

### 3. What did you see instead (Required)
```
id                        	task     	estRows	operator info                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                               	actRows	execution info                                                                                                                                                                                                                 	memory 	disk
	Projection_7              	root     	10     	pacer.pacerq_pinlater_pacer_benchmark_empty_job_tidb_37.local_id, pacer.pacerq_pinlater_pacer_benchmark_empty_job_tidb_37.state, pacer.pacerq_pinlater_pacer_benchmark_empty_job_tidb_37.attempts_allowed, pacer.pacerq_pinlater_pacer_benchmark_empty_job_tidb_37.attempts_remaining, pacer.pacerq_pinlater_pacer_benchmark_empty_job_tidb_37.body, pacer.pacerq_pinlater_pacer_benchmark_empty_job_tidb_37.custom_status, pacer.pacerq_pinlater_pacer_benchmark_empty_job_tidb_37.updated_at, pacer.pacerq_pinlater_pacer_benchmark_empty_job_tidb_37.created_at, pacer.pacerq_pinlater_pacer_benchmark_empty_job_tidb_37.run_after, pacer.pacerq_pinlater_pacer_benchmark_empty_job_tidb_37.server_status	10     	time:3.26s, loops:2, Concurrency:OFF                                                                                                                                                                                           	4.71 KB	N/A
	└─Limit_8                 	root     	10     	offset:0, count:10                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                          	10     	time:3.26s, loops:2                                                                                                                                                                                                            	N/A    	N/A
	  └─TableReader_13        	root     	10     	data:Limit_12                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                               	10     	time:3.26s, loops:1, cop_task: {num: 1, max: 3.26s, proc_keys: 21472, tot_proc: 26.5ms, tot_wait: 134.7µs, rpc_num: 1, rpc_time: 3.26s, copr_cache_hit_ratio: 0.00, build_task_duration: 11.6µs, max_distsql_concurrency: 5} 	1.65 KB	N/A
	    └─Limit_12            	cop[tikv]	10     	offset:0, count:10                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                          	10     	tikv_task:{time:3.26s, loops:25}, scan_detail: {total_process_keys: 21472, total_process_keys_size: 3130080, total_keys: 28623, get_snapshot_time: 8.87µs, rocksdb: {key_skipped_count: 28622, block: {cache_hit_count: 152}}}	N/A    	N/A
	      └─Selection_11      	cop[tikv]	10     	eq(pacer.pacerq_pinlater_pacer_benchmark_empty_job_tidb_37.state, 1), le(pacer.pacerq_pinlater_pacer_benchmark_empty_job_tidb_37.run_after, 2023-04-13 10:24:44.000000)                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                     	112    	tikv_task:{time:3.26s, loops:25}                                                                                                                                                                                               	N/A    	N/A
	        └─TableFullScan_10	cop[tikv]	76.22  	table:pacerq_pinlater_pacer_benchmark_empty_job_tidb_37, keep order:false                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                   	21472  	tikv_task:{time:3.26s, loops:25}                                                                                                                                                                                               	N/A    	N/A
```

### 4. What is your TiDB version? (Required)

<!-- Paste the output of SELECT tidb_version() -->
commit 86f215d8f593996c3593ac866dd2fea2846eda41

### PR DESCRIPTION
**Pull request by time-and-fate**
### What problem does this PR solve?

Issue Number: close #43295

Problem Summary:

Regression in #43295 is introduced by #42921

### What is changed and how it works?

Add a new system variable to control the behavior introduced in #42921, and enable it only for new clusters.

### Check List

Tests 

- [x] Unit test
- [ ] Integration test
- [ ] Manual test (add detailed scripts or steps below)
- [ ] No code

Side effects

- [ ] Performance regression: Consumes more CPU
- [ ] Performance regression: Consumes more Memory
- [ ] Breaking backward compatibility

Documentation

- [ ] Affects user behaviors
- [ ] Contains syntax changes
- [ ] Contains variable changes
- [ ] Contains experimental features
- [ ] Changes MySQL compatibility

### Release note

Please refer to [Release Notes Language Style Guide](https://pingcap.github.io/tidb-dev-guide/contribute-to-tidb/release-notes-style-guide.html) to write a quality release note.

```release-note
None
```

### CODE REVIEW

**Review by qw4990 — APPROVED** [review]

**Review by chrysan — APPROVED** [review]

File: session/bootstrap.go [hunk 1]
```diff
@@ -2506,6 +2510,25 @@ func upgradeToVer143(s Session, ver int64) {
 	doReentrantDDL(s, "ALTER TABLE mysql.tidb_background_subtask ADD COLUMN `error` BLOB", infoschema.ErrColumnExists)
 }
 
+func upgradeToVer144(s Session, ver int64) {
```
[the remark this answers is not in the document]
Comment (time-and-fate) [reply session/bootstrap.go:2513]: added a test case

**Pull request merged**